### DISCUSSION TITLE
Path format, and album vs item values; which one is used?

### DISCUSSION THREAD

**Discussion by bbaserdem** [beetbox/beets#4575] [Q&A] [answered]
I have a flexible tag called `archive` that I apply to either full albums, or individual tracks. The tag shows up under both `Item flexible attributes` and `Album flexible attributes`.

In my path format I want to use this flexible attribute. But there is no explanation on what does the path format read the fields from. For example, both albums and items have the `albumartist` field; while these fields will probably never be different, when I write `$albumartist`  on path format; which data is pulled? I don't know.

My goal is to put things with the `archive` tag in a separate directory. I usually set it for albums; but sometimes I set it for individual tracks. My plan is to add this to the pathspec so that things that are marked as archive are taken to a separate subdirectory. (That subdirectory is on another partition.)

```
%ifdef{archive,%if{${archive},Archive/}}
```

**Comment by JOJ0** [comment]
Interesting question. I'd like to understand that difference as well! I recently had a similar idea: I'd like to move items according to the state of an inline attribute to a separate folder.

**Comment by sampsyo** [comment, marked as answer]
The short answer is that both are used, but album-level data takes priority. The long answer is in the code:
https://github.com/beetbox/beets/blob/d7273bddb34bb247d3ff76f629b8ffa188083cf0/beets/library.py#L371

Namely, when formatting paths, we try the associated album with every item first. If we find no data there, then we look things up in the item itself.

**Comment by wisp3rwind** [comment]
For bespoke path formats, keep the [inline plugin](https://beets.readthedocs.io/en/stable/plugins/inline.html) in mind, too.

**Reply by bbaserdem** [reply]
Thank you. That was my first foray. I used inline plugin, but had an issue so I switched to putting path formats in my own plugin. (Issue was not with the inline plugin, it was that if any of the functions return an empty field, in path format it does not register as an argument to the function, and double directory separators `//` were getting corrected to `_` which messed up my entire library organization.)

I like the write my own plugin approach a bit better; I can format my own python code instead of writing it as a string in `config.yaml`, which makes for better debugging.
